cmake/external/protobuf.cmake

```diff
@@ -0,0 +1,45 @@
+include (ExternalProject)
+
+set(PROTOBUF_INCLUDE_DIRS ${CMAKE_CURRENT_BINARY_DIR}/protobuf/src/protobuf/src)
```

Comment:
This pretty much hard-codes protobuf to be the in-house one; do we want to do something like FindProtobuf.cmake so that when someone is building protobuf, they can use the system installed one?

Reply:
I personally think it's good to use a hard code version and pull it from source. Let me know if you have strong preferences here.